internal/cli/configuration.go

```diff
@@ -69,8 +73,18 @@ var CmdConfigurationCreate = &cobra.Command{
 	Short: "Create a configuration",
 	Long:  `Create configuration by name and key/value dictionary.`,
 	Args: func(cmd *cobra.Command, args []string) error {
-		if len(args) < 3 {
-			return errors.New("Not enough arguments, expected name, key, and value")
+		kvFromFiles, err := cmd.Flags().GetStringSlice("from-file")
+		if err != nil {
+			return errors.Wrap(err, "failed to read option --from-file")
+		}
+		if len(kvFromFiles) == 0 {
+			if len(args) < 3 {
+				return errors.New("Not enough arguments, expected name, key, and value")
+			}
+		} else {
+			if len(args) < 1 {
+				return errors.New("Not enough arguments, expected name")
+			}
```

Comment:
I guess the `name` is always required, and the params still needs to be even, so we can check earlier for these:

```go
if len(args) < 1 {
	return errors.New("Not enough arguments, expected name")
}
if len(args)%2 == 0 {
	return errors.New("Last Key has no value")
}
```

then we can check if no key-val or no files were specified:

```go
if len(kvFromFiles) == 0 && len(args) < 3 {
	return errors.New("Not enough arguments, expected name, key and values, or files.
}
```

Reply:
Ok.